core/controllers/reader.py

```diff
@@ -310,11 +310,16 @@ def post(self, exploration_id):
         old_state_name = self.payload.get('old_state_name')
         # The reader's answer.
         answer = self.payload.get('answer')
+        # The answer handler (submit, click, etc.)
+        handler_name = self.payload.get('handler')
```

Comment:
This also involved removing all invalid uses 'submit' referring to the old handler used for nearly everything.